src/lib/bootloader/grub2_widgets.rb

```diff
@@ -880,16 +883,20 @@ def store
 
       devs = Yast::UI.QueryWidget(:custom_list, :Value)
       devs.split(",").each do |dev|
-        # Add it exactly as specified by the user
-        stage1.add_device(dev.strip)
+        stage1.add_device(DevicePath.new(dev).path)
       end
     end
 
     def validate
       if Yast::UI.QueryWidget(:custom, :Value)
         devs = Yast::UI.QueryWidget(:custom_list, :Value)
         if devs.strip.empty?
-          Yast::Report.Error(_("Custom boot device have to be specied if checked"))
+          Yast::Report.Error(_("Custom boot device has to be specified if checked"))
+          Yast::UI.SetFocus(Id(:custom_list))
+          return false
+        end
+        if !valid_custom_devices?(devs)
+          Yast::Report.Error(_("One of custom partitions is invalid"))
```

Comment:
In fact a little _[Choose...]_ button that would open up a new dialog for selecting a partition from the list does not seem to be that bad idea. In all places where possible. Using the same functionality everywhere.

Reply:
it should be possible. Maybe storage already provide such dialog? I think in general it is not just about partitions, it can be any disk, partition, LVM volume, raid device, etc.